dessia_common/forms.py

```diff
@@ -48,23 +48,43 @@
 from dessia_common.files import BinaryFile, StringFile
 
 
-class StandaloneSubobject(PhysicalObject):
-    _standalone_in_db = True
-    _generic_eq = True
+class EmbeddedBuiltinsSubobject(PhysicalObject):
+    """
+    An object that is not standalone and gather builtins types (float, int, bool, str & Distance).
+
+    :param distarg: A Distance with units
+    :type distarg: Distance
+    :param floatarg: A float
+    :type floatarg: float
+    :param intarg: An integer
+    :type intarg: int
+    :param boolarg: A boolean
+    :type boolarg: bool
+    :param name: Object's name
+    :type name: str
+    """
+    _standalone_in_db = False
 
-    def __init__(self, floatarg: Distance, name: str = 'Standalone Subobject'):
+    def __init__(self, distarg: Distance, floatarg: float, intarg: int,
+                 boolarg: bool, name: str = 'Standalone Subobject'):
+        self.distarg = distarg
         self.floatarg = floatarg
+        self.intarg = intarg
+        self.boolarg = boolarg
 
         PhysicalObject.__init__(self, name=name)
 
     @classmethod
-    def generate(cls, seed: int) -> 'StandaloneSubobject':
-        floatarg = Distance(1.7 * seed)
+    def generate(cls, seed: int) -> 'EmbeddedBuiltinsSubobject':
+        floatarg = 0.3
```

Comment:
You could generate a random number here.
This may avoid some future bugs because a check for a specific float is not as good as with a generic one

Reply:
I think it would be difficult to test it afterwards (we check its value in dessia_common/tests/unit_tests and/or some other test files)